api/src/test/java/com/epam/pipeline/controller/pipeline/PipelineControllerTest.java

```diff
@@ -0,0 +1,827 @@
+                                        BYTE_LIMIT, TEST_INT)));
+        final String contentAsString = mvcResult.getResponse().getContentAsString();
+        final byte[] returnedEntityBody = objectMapper.readValue(contentAsString, byte[].class);
+
+        verify(mockPipelineApiService).getTruncatedPipelineFileContent(ID, TEST_STRING, TEST_STRING, TEST_INT);
+        assertThat(returnedEntityBody).isEqualTo(TEST_ARRAY);
+    }
+
+    @Test
+    public void shouldFailGetTruncatedPipelineFileForUnauthorizedUser() {
+        performUnauthorizedRequest(get(String.format(PIPELINE_ID_FILE_TRUNCATE_URL, ID)));
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldModifyPipelineFile() throws Exception {
+        final String content = getObjectMapper().writeValueAsString(sourceItemVO);
+        doReturn(gitCommitEntry).when(mockPipelineApiService).modifyFile(ID, sourceItemVO);
+
+        final MvcResult mvcResult = performRequest(post(String.format(PIPELINE_ID_FILE_URL, ID)).content(content));
+
+        verify(mockPipelineApiService).modifyFile(ID, sourceItemVO);
+        assertResponse(mvcResult, gitCommitEntry, GitCreatorUtils.GIT_COMMIT_ENTRY_TYPE);
+    }
+
+    @Test
+    public void shouldFailModifyPipelineFileForUnauthorizedUser() {
+        performUnauthorizedRequest(post(String.format(PIPELINE_ID_FILE_URL, ID)));
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldModifyPipelineFiles() throws Exception {
+        final String content = getObjectMapper().writeValueAsString(sourceItemsVO);
+        doReturn(gitCommitEntry).when(mockPipelineApiService).modifyFiles(ID, sourceItemsVO);
+
+        final MvcResult mvcResult = performRequest(post(String.format(PIPELINE_ID_FILES_URL, ID)).content(content));
+
+        verify(mockPipelineApiService).modifyFiles(ID, sourceItemsVO);
+        assertResponse(mvcResult, gitCommitEntry, GitCreatorUtils.GIT_COMMIT_ENTRY_TYPE);
+    }
+
+    @Test
+    public void shouldFailModifyPipelineFilesForUnauthorizedUser() {
+        performUnauthorizedRequest(post(String.format(PIPELINE_ID_FILES_URL, ID)));
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldUploadFile() throws Exception {
+        doReturn(gitCommitEntry).when(mockPipelineApiService).uploadFiles(ID, TEST_STRING, fileMetadataList);
+
+        final MvcResult mvcResult = performRequest(post(String.format(PIPELINE_ID_FILE_UPLOAD_URL, ID))
+                        .params(multiValueMapOf(PATH, TEST_STRING)).content(MULTIPART_CONTENT),
+                MULTIPART_CONTENT_TYPE, EXPECTED_CONTENT_TYPE);
+
+        verify(mockPipelineApiService).uploadFiles(eq(ID), eq(TEST_STRING), anyListOf(UploadFileMetadata.class));
+        final String expectedResult = mvcResult.getResponse().getContentAsString();
+        assertThat(expectedResult).contains(MULTIPART_CONTENT_FILE_NAME)
+                .contains(FILE_SIZE)
+                .contains(MediaType.APPLICATION_OCTET_STREAM_VALUE);
```

Comment:
Could you please use a regular `assertResponse` method? Also let's check that `pipelineApiService.uploadFiles` is called with the proper file metadata using an argument captor.

The controller method is quite an unusual one so let's test most of its logic.